clusterloader2/pkg/measurement/common/system_pod_metrics.go

```diff
@@ -111,7 +168,17 @@ func extractMetrics(lst *v1.PodList) systemPodsMetrics {
 		}
 		metrics.Pods = append(metrics.Pods, podMetrics)
 	}
-	return metrics
+	return &metrics
+}
+
+func buildSummary(podMetrics *systemPodsMetrics) ([]measurement.Summary, error) {
+	content, err := util.PrettyPrintJSON(&podMetrics)
```

Comment:
&podMetrics == **systemPodsMetrics?

Reply:
seems like json encoder will deference arbitrary number of pointers recirsively (ptrEncoder executes typeEncoder on the value that pointer points to...)

https://github.com/golang/go/blob/master/src/encoding/json/encode.go#L432
https://github.com/golang/go/blob/master/src/encoding/json/encode.go#L874-L885

Anyway it is not readable and looks like a bug, I'll fix it